akka-projection-core/src/main/scala/akka/projection/scaladsl/ActorHandler.scala

```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright (C) 2020 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.projection.scaladsl
+
+import scala.util.Try
+
+import akka.Done
+import akka.actor.typed.ActorRef
+import akka.actor.typed.ActorSystem
+import akka.actor.typed.Behavior
+import akka.projection.ProjectionId
+import akka.projection.internal.ActorHandlerImpl
+
+object ActorHandler {
+
+  def apply[Envelope, EnvelopeMessage](
+      projectionId: ProjectionId,
+      behavior: Behavior[_ >: EnvelopeMessage],
```

Comment:
If the user anyway provides a `Behavior` rather than imperative logic to start the actor, I don't quite see why we don't just spawn that as a child of the actor that runs the handler, that would feel natural and nicely namespaced, no risk for leaks etc. Is it because that not always is an actor?

Reply:
That would be nice, but we don't have such a thing in Akka Streams as far as I know. The handler is run by a stream mapAsync.

Reply:
Aha, I see. Is there no actor managing the stream at all?

Reply:
There is an actor running the projection but it's not accessible to the users. We will need to pass it to the projection instance and the projection will need to pass it further to the user handler. 

This will spread logic to places that it makes not much sense, in my opinion. It's better to stay closer to user code here.

Reply:
Now I understand. It's right that there is an actor `ProjectionBehavior` that is running the `Projection`. We could make that integration tighter and then the handler Behavior could be spawned by that actor as a child.

Maybe good to start out with this a documented way of integrating with actors and then we can see if that becomes a popular alternative that we should support more natively. That path means that I must add the `SpawnExtension` and `WatchExtension` to Akka now, but they can be nice additions anyway.

Reply:
@johanandren @renatocaval I have updated this PR based on Johan's idea. The behavior is now spawned as a child of the ProjectionBehavior, which is much better. No need for the spawn and watch extensions.